_docs-2/troubleshooting/code.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Codebase Overview
+category: troubleshooting
+order: 7
+---
+
+If you are trying to troubleshoot an Accumulo problem, you should first try and find help within Accumulo's documentation. The Accumulo
+website's [search tool](https://accumulo.apache.org/search/) can help in finding relevant documentation.
+
+If you cannot find relevant documentation for your problem, you might want to review Accumulo's code to learn more about the code that
+is causing the problem and find a potential solution for it. This documentation provides an overview of Accumulo's codebase to get you
+started.
+
+## Master
+
+The [Master] has the following responsibilities:
+
+  * [detect and respond][update] to [TabletServer] failures
+  * assign and balance tablets to Tablet Servers using a [TabletBalancer]
+  * handle table creation, alteration and deletion requests from clients using the [TableManager]
+  * coordinate changes to write-ahead logs using the [WalStateManager].
+  * report general status
+
+## Metadata Table
```

Comment:
Accumulo has two metadata tables. This section should indicate that `accumulo.metadata` (id: `!0`) contains metadata for user tables, and that `accumulo.root` (id: `+r`) contains metadata for `accumulo.metadata`, and that `ZooKeeper` is used to store metadata for the `accumulo.root` table.